collectors/osd.go

```diff
@@ -0,0 +1,492 @@
+package collectors
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type OsdCollector struct {
+	conn Conn
+
+	CrushWeight *prometheus.GaugeVec
+
+	Depth *prometheus.GaugeVec
+
+	Reweight *prometheus.GaugeVec
+
+	KB *prometheus.GaugeVec
+
+	UsedKB *prometheus.GaugeVec
+
+	AvailKB *prometheus.GaugeVec
+
+	Utilization *prometheus.GaugeVec
+
+	Pgs *prometheus.GaugeVec
+
+	CommitLatency *prometheus.GaugeVec
+
+	ApplyLatency *prometheus.GaugeVec
+
+	OsdIn *prometheus.GaugeVec
+
+	OsdUp *prometheus.GaugeVec
+
+	TotalKB prometheus.Gauge
+
+	TotalUsedKB prometheus.Gauge
+
+	TotalAvailKB prometheus.Gauge
+
+	AverageUtil prometheus.Gauge
+}
+
+func NewOsdCollector(conn Conn) *OsdCollector {
+	return &OsdCollector{
+		conn: conn,
+
+		CrushWeight: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_crush_weight",
+				Help:      "OSD Crush Weight",
+			},
+			[]string{"osd"},
+		),
+
+		Depth: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_depth",
+				Help:      "OSD Depth",
+			},
+			[]string{"osd"},
+		),
+
+		Reweight: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_reweight",
+				Help:      "OSD Reweight",
+			},
+			[]string{"osd"},
+		),
+
+		KB: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: cephNamespace,
+				Name:      "osd_kb",
```

Comment:
According to the convention we observe: https://prometheus.io/docs/practices/naming/ these should be specified in `_bytes`. I see that the values we get directly from Ceph as in `KBs` but it would help us more to be consistent throughout and handle the resolution (if we want to switch to KBs or higher) on the viewer side.